streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -639,7 +666,20 @@ abstract class DStream[T: ClassTag] (
     // because the DStream is reachable from the outer object here, and because
     // DStreams can't be serialized with closures, we can't proactively check
     // it for serializability and so we pass the optional false to SparkContext.clean
-    new ForEachDStream(this, context.sparkContext.clean(foreachFunc, false)).register()
+    foreachRDD(foreachFunc, makeRDDOpsVisible = true)
+  }
+
+  /**
+   * Apply a function to each RDD in this DStream. This is an output operator, so
+   * 'this' DStream will be registered as an output stream and therefore materialized.
+   * Setting `makeRDDOpsVisible` will allow the callsites and scopes of the RDDs generated
+   * in the `foreachFunc` to be visible in the UI.
+   */
+  private[streaming] def foreachRDD(
+      cleanedForeachFunc: (RDD[T], Time) => Unit,
+      makeRDDOpsVisible: Boolean): Unit = {
```

Comment:
I'm wondering if we really need this method. Looks `makeRDDOpsVisible` is always `true`.